### ISSUE TITLE
Bug in TuttePolynomial

### ISSUE THREAD

**Issue by mohamed-barakat** [martin-leuner/alcove#8]
Commit dc4ba73 introduced the following bug in `TuttePolynomial`:

```
gap> LoadPackage( "alcove" );
true
gap>
gap> F3 := HomalgRingOfIntegers( 3 );
GF(3)
gap> mat :=
>   [ [ Z(3)^0, 0*Z(3), Z(3)^0, 0*Z(3), Z(3)^0, Z(3), Z(3), Z(3)^0, 0*Z(3) ],
>     [ 0*Z(3), Z(3)^0, Z(3), 0*Z(3), Z(3)^0, Z(3), 0*Z(3), Z(3), Z(3)^0 ],
>     [ 0*Z(3), 0*Z(3), 0*Z(3), Z(3)^0, Z(3)^0, Z(3)^0, Z(3)^0, Z(3)^0, Z(3)^0 ] ];
[ [ Z(3)^0, 0*Z(3), Z(3)^0, 0*Z(3), Z(3)^0, Z(3), Z(3), Z(3)^0, 0*Z(3) ],
  [ 0*Z(3), Z(3)^0, Z(3), 0*Z(3), Z(3)^0, Z(3), 0*Z(3), Z(3), Z(3)^0 ], [ 0*Z(3), 0*Z(3), 0*Z(3), Z(3)^0, Z(3)^0, Z(3)^0, Z(3)^0, Z(3)^0,
      Z(3)^0 ] ]
gap> mat := HomalgMatrix( mat, F3 );
<A matrix over an internal ring>
gap> mat := Matroid( mat );
<A vector matroid>
gap> TuttePolynomial( mat );
Error, no method found! For debugging hints type ?Recovery from NoMethodFound
Error, no 3rd choice method found for `TuttePolynomial' on 3 arguments called from
TuttePolynomial( matroid, xy[1], xy[2] ) called from
<function "unknown">( <arguments> )
 called from read-eval loop at line 10 of *stdin*
you can 'quit;' to quit to outer loop, or
you can 'return;' to continue
brk>
gap> TuttePolynomial( mat );
y^6+3*y^5+6*y^4+x^3+10*y^3+6*x^2+12*x*y+15*y^2+9*x+9*y
```

**Comment by martin-leuner** [comment]
I see. I have a fix in mind, but it'll need some thorough testing.

**Comment by martin-leuner** [comment]
Should be fixed in a80ca92321e4d82909c654095bbe3eb188ae7487.
